api/resolver/v1/resolver_api.proto

```diff
@@ -34,6 +34,34 @@ service ResolverAPI {
     };
     option (clutch.api.v1.action).type = READ;
   }
+  rpc Autocomplete(AutocompleteRequest) returns (AutocompleteResponse) {
+    option (google.api.http) = {
+      post : "/v1/resolver/autocomplete"
+      body : "*"
+    };
+    option (clutch.api.v1.action).type = READ;
+  }
+}
+
+message AutocompleteRequest {
+  // The type URL of the desired result.
+  string want = 1 [ (validate.rules).string = {min_bytes : 1} ];
+  string search = 2 [ (validate.rules).string = {min_bytes : 1} ];
+  uint64 limit = 3;
+}
+
+message AutocompleteResponse {
+  message AutocompleteResult {
```

Comment:
super nit: if you think you'll need `AutocompleteResult` elsewhere, may want to move it out `AutocompleteResponse`

Reply:
Yeah i was thinking about this, maybe i'll just move it out just incase? But nothing comes to mind at the moment.

Reply:
could be useful to move it out if we want to use it as a type on the frontend